src/Microsoft.Graph.Core/Requests/HttpProvider.cs

```diff
@@ -244,15 +244,9 @@ public Task<HttpResponseMessage> SendAsync(HttpRequestMessage request)
             {
                 return await this.httpClient.SendAsync(request, completionOption, cancellationToken).ConfigureAwait(false);
             }
-            catch (TaskCanceledException exception)
+            catch (OperationCanceledException)
             {
```

Comment:
why not just remove the catch all together since nothing is done with the exception?

Reply:
There is a final `catch` block that will catch all derived types of `Exception` and still wrap it in a ServiceException which is contrary to our desire.